### PR TITLE
GW-28: Refactor button component

### PR DESCRIPTION
**Pull request by Julian-UBC**
- Created a template for button
- Made three options for button: solid background button, transparent background button, landing image button
- Refactor all components to use the new button template

Preview: https://deploy-preview-16--gisaubcv3.netlify.app/

### CODE REVIEW

File: src/pages/home/components/LandingImage.js [hunk 1]
```diff
@@ -27,7 +27,11 @@ export const LandingImage = () => {
             Gado-Gado Indonesian Student Association <br />
             of UBC
           </h1>
-          <CustomButton className="grid justify-center" />
+          <Button
+            className="grid justify-center"
+            text="Become a member!"
+            landingButton={true}
```
Comment (chrishadrian) [inline src/pages/home/components/LandingImage.js:33]: Why don't we just directly use the landingButton component?

File: src/shared/components/Button/Button.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import "./Button.css";
+
+export const Button = ({
+  landingButton,
+  className,
+  text,
+  transparentBg,
+  handleClickButton,
+}) => {
+  if (landingButton) {
```
Comment (chrishadrian) [inline src/shared/components/Button/Button.js:13]: Yeah I think this is so redundant. Imo, it's better to use the landingButton component directly as stated above.

File: src/shared/layout/EventLayout.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ export const EventLayout = ({
           <div className="event-img">{event.img}</div>
         </div>
         <div className="grid justify-center">
-          <CustomButton name={button1} />
+          <Button text={button1} />
```
Comment (chrishadrian) [inline src/shared/layout/EventLayout.js:32]: 1. The buttons aren't aligned to the center. 
2. The width difference between button1 and button2 is quite significant. Can you make them to have a fixed width instead (referring to figma)?
image

**Review by chrishadrian — APPROVED** [review]
Very good job on this one Jul! I just added some minor comments. Cheers.

**Review by jenaarianto — APPROVED** [review]

File: src/pages/home/components/LandingImage.js (hunk 1 above)
Comment (Julian-UBC) [reply src/pages/home/components/LandingImage.js:33]: I want to make every button to use the Button component so its more uniform

File: src/shared/components/Button/Button.js (hunk 2 above)
Comment (Julian-UBC) [reply src/shared/components/Button/Button.js:13]: It is a bit redundant. But as stated in previous comment.

File: src/shared/layout/EventLayout.js (hunk 3 above)
Comment (Julian-UBC) [reply src/shared/layout/EventLayout.js:32]: Will incorporate this in a new PR!

**Pull request merged**